server/game/gameSystems/SearchDeckSystem.ts

```diff
@@ -144,17 +143,20 @@ export class SearchDeckSystem extends PlayerTargetSystem<ISearchDeckProperties>
         let selectAmount = 1;
         const choosingPlayer = properties.choosingPlayer || event.player;
 
-        if (properties.targetMode === TargetMode.UpTo || properties.targetMode === TargetMode.UpToVariable) {
-            selectAmount = this.getNumCards(properties.selectCount, context);
-        }
-        if (properties.targetMode === TargetMode.Single) {
-            selectAmount = 1;
-        }
-        if (properties.targetMode === TargetMode.Exactly || properties.targetMode === TargetMode.ExactlyVariable) {
-            selectAmount = this.getNumCards(properties.selectCount, context);
-        }
-        if (properties.targetMode === TargetMode.Unlimited) {
-            selectAmount = -1;
+        switch (properties.targetMode) {
+            case TargetMode.UpTo:
+            case TargetMode.UpToVariable:
+            case TargetMode.Exactly:
+            case TargetMode.ExactlyVariable:
+                selectAmount = this.getNumCards(properties.selectCount, context);
+                break;
+            case TargetMode.Unlimited:
+                selectAmount = -1;
+                break;
+            case TargetMode.Single:
+            default:
```

Comment:
One very small favor to ask - if the above switch cases grab all of the expected values, can we repeat the pattern I've used elsewhere which is throwing an assertion failure in the default case:

```
default:
    Contract.fail(`Invalid targetMode: ${properties.targetMode}`);
    selectAmount = -1; // or some other useful default value that will let things try to move forward
    break;
```

This type of contract check probably catches the most errors of any of them in my work on the repo.